src/credential/credentialController.ts

```diff
@@ -180,33 +181,32 @@ export class CredentialController {
             state,
         });
 
-        try {
-            let opened = await vscode.env.openExternal(vscode.Uri.parse(signUrl));
-            if (opened) {
-                let result = await handleAuthCallback(async (uri: vscode.Uri, resolve: (result: UserInfo) => void, reject: (reason: any) => void) => {
-                    try {
-                        const query = parseQuery(uri);
-
-                        resolve({
-                            userName: query.name,
-                            userEmail: query.email,
-                            userToken: query['X-OP-BuildUserToken'],
-                            signType: 'GitHub'
-                        });
-                    } catch (err) {
-                        reject(err);
-                    }
+        let opened = await vscode.env.openExternal(vscode.Uri.parse(signUrl));
+        if (!opened) {
+            // User decline to open external URL to sign in
+            throw new DocsError(`Sign-in with GitHub Failed: Please Allow Code to open `, ErrorCode.GitHubSignInExternalUrlDeclined);
+        }
+
+        let result = await handleAuthCallback(async (uri: vscode.Uri, resolve: (result: UserInfo) => void, reject: (reason: any) => void) => {
+            try {
+                const query = parseQuery(uri);
+
+                resolve({
+                    userName: query.name,
+                    userEmail: query.email,
+                    userToken: query['X-OP-BuildUserToken'],
+                    signType: 'GitHub'
                 });
-                if (result instanceof Error) {
-                    throw result;
-                }
-                return result;
+            } catch (err) {
+                reject(err);
             }
-            this.eventStream.post(new UserSignInFailed(`Sign-in with GitHub Failed`));
-            return undefined;
-        } catch (err) {
-            this.eventStream.post(new UserSignInFailed(`Sign-in with GitHub Failed: ${err.message}`));
-            return undefined;
+        });
+        if (result instanceof Error) {
+            if (result instanceof TimeOutError) {
+                throw new DocsError(`Sign-in with GitHub Failed: Time out`, ErrorCode.GitHubSignInTimeOut, result);

```

Comment:
should `Time out` be the message of `TimeOutError`?
Then the logic can be simplified as ``throw new DocsError(`Sign-in with GitHub Failed: ${result.message}`,  result instanceof TimeOutError ? ErrorCode.GitHubSignInTimeOut : ErrorCode.GitHubSignInFailed, result);``